pkg/policy/validate_controller.go

```diff
@@ -197,7 +197,7 @@ func (pc *PolicyController) addPolicy(obj interface{}) {
 
 	logger.Info("policy created", "uid", p.UID, "kind", "ClusterPolicy", "name", p.Name)
 
-	if p.Spec.Background == nil || p.Spec.ValidationFailureAction == "" || checkAutoGenRules(p) {
+	if p.Spec.Background == nil || p.Spec.ValidationFailureAction == "" || checkKind(p) || checkAutoGenRules(p) {
```

Comment:
Where did this message print out?
>The ClusterPolicy "disallow-bind-mounts" is invalid: spec.rules.match.resources.kinds: Invalid value: "null": spec.rules.match.resources.kinds

We cannot add this check in the background controller, it should happen in the webhook https://github.com/kyverno/kyverno/blob/main/pkg/webhooks/policyvalidation.go.

And how do handle backward compatibility?

Reply:
I will make changes according to suggestions